gym_cap/envs/cap_env.py

```diff
@@ -493,6 +493,11 @@ def step(self, entities_action=None, cur_suggestions=None):
         self.state = np.copy(self._env)
 
         isDone = self.red_win or self.blue_win
+        
+        for agent in self.team_blue + self.team_red:
+            if agent.memory_mode == "fog":
+                Agent.update_memory()
```

Comment:
Agent.update_memory causes an issue.
update_memory needs to be called for individual agents.

Also, please pass the environment parameter.
```py
agent.update_memory(env=self)
```